py/dml/codegen.py

```diff
@@ -1573,10 +1573,19 @@ def stmt_log(stmt, location, scope):
         if location.method():
             key = mkLit(site, '(uint64)NULL', TInt(64, False))
         else:
-            # Normally, we are not allowed to cast "this.trait" to an uint,
-            # but since we _internally_ know it is a pointer this is justified
-            key = mkLit(site, "(uint64)(%s.trait)" % lookup_var(
-                site, scope, "this").read(), TInt(64, False))
+            # Acquire a key based on trait _identity_t
+            identity = TraitObjIdentity(site, lookup_var(site, scope, "this"))
+            # identity is opaque, so we need to cheat a little bit
```

Comment:
An alternative is to do the necessary manipulation through a method in `dmllib.h` instead.

Reply:
This seems reasonable. It puts the conversion function, which depends on representation defined in dmllib, in the same place